### PR TITLE
Migrate PathOps Lib to VectorGraphics

### PR DESCRIPTION
**Pull request by c-h-i-a-m-a-k-a-2**
- Integrates the PathOps Lib that normally exists in flutter/engine into Vector Graphics Library
- Allows for easier use by MaskingOptimizer and ClippingOptimizer
- Removes the need for hard-coded paths, and allows for Optimizers to not have to use local engines

### CODE REVIEW

File: packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart [hunk 1]
```diff
@@ -15,11 +15,22 @@ final ArgParser argParser = ArgParser()
     valueHelp: 'path/to/libtessellator.dylib',
     hide: true,
   )
+  ..addOption(
+    'libpathops',
+    help: 'The path to a libpathops dynamic library',
+    valueHelp: 'path/to/libpathops.dylib',
```
Comment (dnfield) [inline packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart:21]:
```suggestion
    valueHelp: 'path/to/libpath_ops.dylib',
```

File: packages/vector_graphics_compiler/lib/src/initialize_path_ops.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  } else {
+    print('Unknown executable: ${Platform.resolvedExecutable}');
+    return false;
+  }
+
+  final String platform;
+  final String executable;
+  if (Platform.isWindows) {
+    platform = 'windows-x64';
+    executable = 'libpathops.dll';
```
Comment (dnfield) [inline packages/vector_graphics_compiler/lib/src/initialize_path_ops.dart:20]:
```suggestion
    executable = 'path_ops.dll';
```

File: packages/vector_graphics_compiler/lib/src/initialize_path_ops.dart [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  } else if (Platform.isMacOS) {
+    platform = 'darwin-x64';
+    executable = 'libpathops.dylib';
```
Comment (dnfield) [inline packages/vector_graphics_compiler/lib/src/initialize_path_ops.dart:23]:
```suggestion
    executable = 'libpath_ops.dylib';
```

File: packages/vector_graphics_compiler/lib/src/initialize_path_ops.dart [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  } else if (Platform.isLinux) {
+    platform = 'linux-x64';
+    executable = 'libpathops.so';
```
Comment (dnfield) [inline packages/vector_graphics_compiler/lib/src/initialize_path_ops.dart:26]:
```suggestion
    executable = 'libpath_ops.so';
```

File: packages/vector_graphics_compiler/lib/src/svg/path_ops.dart [hunk 5]
```diff
@@ -0,0 +1,390 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
```
Comment (dnfield) [inline packages/vector_graphics_compiler/lib/src/svg/path_ops.dart:4]: Please add a comment about where this was copied from and that it should be kept in sync with that source for now.

**Review by dnfield — COMMENTED** [review]
Can you add the tests for path_ops here too? It'd be good to see them running here.

File: packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart [hunk 6]
```diff
@@ -15,11 +15,22 @@ final ArgParser argParser = ArgParser()
...
+    hide: true,
+  )
   ..addFlag(
     'tessellate',
     help: 'Convert path fills into a tessellated shape. This will improve '
         'raster times at the cost of slightly larger file sizes.',
   )
+  ..addFlag(
+    'pathops',
```
[the remark this answers is not in the document]
Comment (c-h-i-a-m-a-k-a-2) [reply packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart:30]: fixed.

File: packages/vector_graphics_compiler/lib/src/svg/path_ops.dart (hunk 5 above)
Comment (c-h-i-a-m-a-k-a-2) [reply packages/vector_graphics_compiler/lib/src/svg/path_ops.dart:6]: Added.

**Comment by c-h-i-a-m-a-k-a-2** [comment]
> Can you add the tests for path_ops here too? It'd be good to see them running here.

Added test file.

**Comment by c-h-i-a-m-a-k-a-2** [comment]
> Can you take a look at CI and fix up the analysis issues?

done.

**Pull request merged**